### PR TITLE
Table sort column APIs

### LINKED ISSUE #5906: [Feature] Table widget: Sort order API
## Summary

As a dev, I should have access to the table sort column if any through the `{{table.sortOrder}}` API. This will return an object when sort is active > `{column: string, order: enum|‘asc’|'desc'}`. There should also be a `onSortChange` action available which can trigger an API

## Motivation

Devs would like to take control of sorting in the API while doing server side pagination in table widget

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Sorted column can be accessed using sortOrder table widget api. Add onSort action that lets user trigger an action whenever column sorting is changed

Fixes #5906 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feature/table-sort-column-properties 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.62 **(0)** | 36.65 **(-0.03)** | 33.05 **(0.01)** | 55.23 **(0.01)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/Constants.ts | 85.48 **(0.73)** | 73.53 **(1.65)** | 91.3 **(0.39)** | 85.48 **(0.73)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 37.72 **(-0.88)** | 36.21 **(0)** | 55.08 **(-0.27)**
 :red_circle: | app/client/src/widgets/TableWidget/derived.js | 54.26 **(0)** | 53.57 **(-2.68)** | 37.84 **(0)** | 54.07 **(0)**
 :green_circle: | app/client/src/widgets/TableWidget/index.tsx | 27.71 **(0.07)** | 6.25 **(-0.04)** | 25.49 **(0)** | 27.9 **(0.08)**</details>

### CODE REVIEW

**Comment by vivekverma2312** [comment]
@vicky-primathon Functionality is working fine but can you please add tooltip for sort property.

File: app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts [hunk 1]
```diff
@@ -756,6 +756,7 @@ export default [
         isTriggerProperty: true,
       },
       {
+        helpText: "Triggers an action when a table is sorted on a column",
```
Comment (somangshu) [inline app/client/src/widgets/TableWidget/TablePropertyPaneConfig.ts:759]: Change to "Triggers an action when a table column is sorted"

**Review by somangshu — CHANGES_REQUESTED** [review]

**Review by SatishGandham — APPROVED** [review]

**Comment by somangshu** [comment]
@vicky-primathon can you merge the latest release as well and check for the the failing test cases?

**Review by somangshu — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@vicky-primathon Tooltip is also fixed.

**Comment by somangshu** [comment]
Rerunning all tests, Unrelated failures found. 

@vicky-primathon please assert.

**Review by somangshu — APPROVED** [review]

**Review by somangshu — DISMISSED** [review]
Approving to run tests

File: app/client/src/utils/autocomplete/EntityDefinitions.ts [hunk 2]
```diff
@@ -70,6 +70,11 @@ export const entityDefinitions = {
     pageSize: "number",
     isVisible: isVisible,
     searchText: "string",
+    sortOrder: {
+      column: "string",
```
Comment (SatishGandham) [inline app/client/src/utils/autocomplete/EntityDefinitions.ts:74]: Sort order should be a enum, not boolean.
['ASC','DESC']

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Comment by YogeshJayaseelan** [comment]
@vicky-primathon Testing in-progress for this feature and please find the observations below

1. asc:true or asc:false key and value should not be displayed when the table column is bind-ed with text widget 

Text Widget - Label field - {{JSON.stringify(Table1.sortOrder)}}

Screenshot 2021-08-13 at 4 58 01 PM

2.By default when the table's column and order is sorted and stored in a text widget - 'undefined, undefined' key and value is displayed 

Screenshot 2021-08-13 at 5 02 15 PM

**Comment by YogeshJayaseelan** [comment]
@vicky-primathon Verified this PR and it is working as expected

**Review by SatishGandham — APPROVED** [review]

**Review by somangshu — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@vicky-primathon Issue 1 and 2 now exists in release environment

**Pull request merged**